ironfish/src/primitives/target.ts

```diff
@@ -140,16 +141,39 @@ export class Target {
 
     // max(1 - (current_block_timestamp - parent_timestamp) / 10, -99)
     const diffInSeconds = (time.getTime() - previousBlockTimestamp.getTime()) / 1000
-    const sign = BigInt(Math.max(1 - Math.floor(diffInSeconds / 10), -99))
-    const offset = BigInt(previousBlockDifficulty) / BigInt(DIFFICULTY_ADJUSTMENT_DENOMINATOR)
 
     // diff = parent_diff + parent_diff / 2048 * max(1 - (current_block_timestamp - parent_timestamp) / 10, -99)
-    const difficulty = BigIntUtils.max(
-      BigInt(previousBlockDifficulty) + offset * sign,
-      Target.minDifficulty(),
-    )
 
-    return difficulty
+    const threshold = 1
+    const bucket = this.findBucket(diffInSeconds)
+    const maxChange = previousBlockDifficulty / BigInt(DIFFICULTY_ADJUSTMENT_DENOMINATOR)
```

Comment:
why can't we just put 2048 here? is DIFFICULTY_ADJUSTMENT_DENOMINATOR ever used anywhere else?

Reply:
also again, `maxChange` implies that we'll be changing the difficulty _maximum_ that number, which isn't true